relayer/chains/wasm/provider.go

```diff
@@ -318,79 +317,52 @@ func (p *Provider) pollTxResultStream(ctx context.Context, txHash string, maxWai
 	return txResChan
 }
 
-func (p *Provider) subscribeTxResultStream(ctx context.Context, txHash string, maxWaitInterval time.Duration) <-chan *types.TxResultChan {
-	txResChan := make(chan *types.TxResultChan)
-	go func(txRes chan *types.TxResultChan) {
-		defer close(txRes)
+func (p *Provider) subscribeTxResult(ctx context.Context, tx *sdkTypes.TxResponse, maxWaitInterval time.Duration) (*types.TxResult, error) {
+	newCtx, cancel := context.WithTimeout(ctx, maxWaitInterval)
```

Comment:
Any reason why this part is sync now ? any impact it can have or have you tested all scenarios to ensure it won't have any impact moving from async to sync ?

Reply:
It's same as with other chains implementation now, spinning go routine is unnecessary as wait for tx result is blocking and parent call is already on routine.